### PR TITLE
Add registerApi function

### PR DESCRIPTION
**Pull request by coderkevin**
This adds the registerApi function along with validation and tests.

To Test:

1. `npm test` and ensure all tests pass.

### CODE REVIEW

File: src/registry/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import isApiValid from '../utils/is-api-valid';
+
+const debug = debugFactory( 'fresh-data:registry' );
+const _apis = new Map();
+
+export function registerApi( api, apis = _apis ) {
+	debug( `Registering api "${ api && api.name }"` );
+
+	if ( ! isApiValid( api ) ) {
+		throw new Error( 'Invalid API object' );
```
Comment (timmyc) [inline src/registry/index.js]: Should the error message be a bit more descriptive about _why_ the API object was invalid? Just thinking that might make it easier for 3pd to decipher what went wrong.

Comment (timmyc) [reply src/registry/index.js]: I see it is output as a debug, disregard.

**Comment by timmyc** [comment]
```
➜  fresh-data git:(add/api-registry) npm test

> fresh-data@0.1.0 test /Users/timmyc/code/fresh-data
> cross-env BABEL_ENV=test NODE_ENV=test jest --runInBand --coverage

2018-04-10T15:38:23,450: [2324194176] the owner of /usr/local/Cellar/watchman/HEAD/var/run/watchman/timmyc-state is uid 0 and doesn't match your euid 501

Watchman:  watchman --no-pretty get-sockname returned with exit code=1, signal=null, stderr= 2018-04-10T15:38:23,450: [2324194176] the owner of /usr/local/Cellar/watchman/HEAD/var/run/watchman/timmyc-state is uid 0 and doesn't match your euid 501
```

I get this when attempting to run `npm test`. My magical breaking skills are intact!

**Comment by coderkevin** [comment]
> I get this when attempting to run npm test. My magical breaking skills are intact!

I'm not sure how you're getting this error, but it looks like something that was installed by jest. I think I may have found a workaround in the last comment on this GH issue: https://github.com/facebook/react-native/issues/2941

> In my case (watchman 4.9.0),it show me the error: watchman --no-pretty get-sockname returned with exit code=1, signal=null, stderr= 2018-03-08T09:53:11,047: [0x7fffac994340] the owner of /usr/local/var/run/watchman/root-state is uid 501 and doesn't match your euid 0
>
> I try all above possible way but no luck and the only way to fix is going to
the dir of /usr/local/var/run/watchman/ , and delete the dir of root-state, then restart watchman.

**Comment by timmyc** [comment]
Thanks for the permissions idea, that fixed it up @coderkevin - all green now

**Pull request merged**